oauth2server/apps/accounts/views.py

```diff
@@ -81,10 +80,28 @@ def put(self, request):
             # need to tell the other system we are not capable of creating the resource.
             raise DatabaseFailureException
 
+        # if account_locked attribute set to False, we want to unlock the account
+        try:
+            # Check to see if the POST contains the account_locked flag.
+            # If so we unlock account and reset failed log-ins.
+            if request.POST.get('account_locked') == 'False':
+                stdlogger.debug('Admin is unlocking account')
+                request.user.unlock_account()
+                request.user.reset_failed_logins()
+                # Verify user if not verified
+                if not request.user.account_is_verified:
```

Comment:
Not sure this is right. What if an unverified account is locked? if you unlock it you wouldn't want to automatically verify the account as well.

Reply:
We have had approval from the business that this is the behaviour we want, as we are happy at this point they have either:

- access their email to reset (i.e. unlock)
- Called up internal and verified over the phone